### PR TITLE
Add main CI workflow to GH Actions

### PR DESCRIPTION
**Pull request by epwalsh**
Moves main CI workflow to GitHub Actions and also runs model tests (closes https://github.com/allenai/allennlp/issues/4080).

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python: ['3.6', '3.7']
+
+    steps:
+    - uses: actions/checkout@v1
```
Comment (bryant1410) [inline .github/workflows/main.yml:19]:
```suggestion
    - uses: actions/checkout@v2
```

It's slightly faster :smile:

**Comment by epwalsh** [comment]
Still TODO:
- [ ] Add release workflow
- [ ] Add Docker workflow
- [ ] Add step to deploy master and release docs

File: .github/workflows/checks.yml [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    - name: Debug info
+      run: |
+        pip freeze
+
+    - name: Pull and install models repo
+      env:
+        EXCLUDE_ALLENNLP_IN_SETUP: "true"
+      run: |
+        git clone https://github.com/allenai/allennlp-models.git
+        cd allennlp-models && pip install -e . && pip install -r dev-requirements.txt
```
Comment (dirkgr) [inline .github/workflows/checks.yml:103]: Wow, that was way easier than I feared.

File: Makefile [hunk 3]
```diff
@@ -20,6 +20,35 @@ else
...
+.PHONY : test
+test :
+	pytest --color=yes -rf --durations=40 -k "not sniff_test" $(SRC)
+
+.PHONY : test-with-cov
+test-with-cov :
+	pytest --color=yes -rf --cov-config=.coveragerc --cov=$(SRC) --durations=40 -k "not sniff_test" $(SRC)
+
+#
+# Documention helpelrs.
```
Comment (dirkgr) [inline Makefile:49]: "helpers"

File: .github/workflows/checks.yml [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+  push:
+    branches:
+    - master
+
+jobs:
+  core:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python: ['3.6', '3.7']
```
Comment (bryant1410) [inline .github/workflows/checks.yml:20]: A bit late, but why not adding 3.8?

**Pull request merged**